resources/dw/virtualwarehouse/impala/resource_impala_vw.go

```diff
@@ -0,0 +1,262 @@
+	return func() (any, string, error) {
+		tflog.Debug(ctx, "About to describe Impala")
+		params := operations.NewDescribeVwParamsWithContext(ctx).
+			WithInput(&models.DescribeVwRequest{ClusterID: clusterID, VwID: vwID})
+		resp, err := r.client.Dw.Operations.DescribeVw(params)
+		if err != nil {
+			if strings.Contains(err.Error(), "Virtual Warehouse not found") {
+				return &models.DescribeVwResponse{}, "Deleted", nil
+			}
+			*callFailedCount++
+			if *callFailedCount <= callFailureThreshold {
+				tflog.Warn(ctx, fmt.Sprintf("could not describe Data Warehouse Impala Virtual Warehouse "+
+					"due to [%s] but threshold limit is not reached yet (%d out of %d).", err.Error(), callFailedCount, callFailureThreshold))
+				return nil, "", nil
+			}
+			tflog.Error(ctx, fmt.Sprintf("error describing Data Warehouse Impala Virtual Warehouse due to [%s] "+
+				"failure threshold limit exceeded.", err.Error()))
+			return nil, "", err
+		}
+		*callFailedCount = 0
+		vw := resp.GetPayload()
+		tflog.Debug(ctx, fmt.Sprintf("Described Impala %s with status %s", vw.Vw.ID, vw.Vw.Status))
+		return vw, vw.Vw.Status, nil
+	}
+}
+
+func createVwRequestFromPlan(plan *resourceModel) *models.CreateVwRequest {
+	req := &models.CreateVwRequest{
+		Name:      plan.Name.ValueStringPointer(),
+		ClusterID: plan.ClusterID.ValueStringPointer(),
+		DbcID:     plan.DatabaseCatalogID.ValueStringPointer(),
+		VwType:    models.VwTypeImpala.Pointer(),
+	}
+	if imageVersion := plan.ImageVersion.ValueString(); imageVersion != "" {
+		req.ImageVersion = imageVersion
+	}
+	return req
+}
+
+func (r *impalaResource) waitForVwRunning(ctx context.Context, plan *resourceModel, payload *models.CreateVwResponse) error {
+	clusterID := plan.ClusterID.ValueStringPointer()
+	vwID := &payload.VwID
+
+	opts := plan.PollingOptions
+
+	if opts == nil || !opts.Async.ValueBool() {
+		callFailedCount := 0
+		stateConf := &retry.StateChangeConf{
+			Pending:      []string{"Accepted", "Creating", "Created", "Starting"},
+			Target:       []string{"Running"},
+			Delay:        30 * time.Second,
+			Timeout:      utils.GetPollingTimeout(plan, 20*time.Minute),
+			PollInterval: 30 * time.Second,
+			Refresh:      r.stateRefresh(ctx, clusterID, vwID, &callFailedCount, utils.GetCallFailureThreshold(plan, 3)),
+		}
+		_, err := stateConf.WaitForStateContext(ctx)
+		return err
+	}
+	return nil
+}
+
+func (r *impalaResource) populatePlanFromDescribe(ctx context.Context, plan *resourceModel, vwID *string) error {
+	desc := operations.NewDescribeVwParamsWithContext(ctx).
+		WithInput(&models.DescribeVwRequest{VwID: vwID, ClusterID: plan.ClusterID.ValueStringPointer()})
+	describe, err := r.client.Dw.Operations.DescribeVw(desc)
+	if err != nil {
+		return err
+	}
+
+	impala := describe.GetPayload()
+	plan.ID = types.StringValue(impala.Vw.ID)
```

Comment:
You can create a method on resourceModel to populate the plan data, you can leave then this piece in the caller function.

Reply:
Done !